File: updateEnvs.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Collections;
using System.Net.Http.Headers;


string sourceUrl = "";
string sourceClientID = "";
string sourceClientSecret = "";

string distUrl = "";
string distClientID = "";
string distClientSecret = "";


string sourceToken = await GetTokenAsync(sourceUrl, "", "");
JArray? cookieArray = await GetEnvsAsync(sourceUrl, sourceToken);

ArrayList cookies = new ArrayList();
foreach (JObject item in cookieArray)
{
    cookies.Add(item["value"].ToString());
}
string cookieStr = string.Join("&", (string[])cookies.ToArray(typeof(string)));

string distToken = await GetTokenAsync(distUrl, "", "");
JArray? cookiesInfo = await GetEnvsAsync(distUrl, distToken);

int id = 0;
foreach (JObject item in cookiesInfo)
{
    if (item["name"].ToString() == "JD_COOKIE")
    {
        id = Convert.ToInt32(item["id"].ToString());
    }
}

JObject? res = await UpdateEnvsAsync(new QL(id, distUrl, distToken, cookieStr));

if (Convert.ToInt32(res["code"].ToString()) == 200)
{
    Console.WriteLine("更新JD_COOKIE环境变量成功!");
}
else
{
    Console.WriteLine("更新JD_COOKIE环境变量失败!");
}

Console.ReadLine();


static async Task<string> GetTokenAsync(string url, string client_id, string client_secret)
{
    url = $"{url}/open/auth/token?client_id={client_id}&client_secret={client_secret}";
    using var httpClient = new HttpClient();
    using var request = new HttpRequestMessage(new HttpMethod("Get"), url);
    var response = await httpClient.SendAsync(request);
    using HttpContent content = response.Content;
    JObject? res = JsonConvert.DeserializeObject(await content.ReadAsStringAsync()) as JObject;
    return $"{res["data"]["token_type"]} {res["data"]["token"]}";
}


static async Task<JArray?> GetEnvsAsync(string url, string token)
{
    url = $"{url}/open/envs";
    using var httpClient = new HttpClient();
    using var request = new HttpRequestMessage(new HttpMethod("Get"), url);
    request.Headers.TryAddWithoutValidation("Accept", "application/json");
    request.Headers.TryAddWithoutValidation("Authorization", token);
    var response = await httpClient.SendAsync(request);
    using HttpContent content = response.Content;
    JObject? res = JsonConvert.DeserializeObject(await content.ReadAsStringAsync()) as JObject;
    return res["data"] as JArray;
}


static async Task<JObject?> UpdateEnvsAsync(QL ql)
{
    string url = $"{ql.url}/open/envs";
    using var httpClient = new HttpClient();
    using var request = new HttpRequestMessage(new HttpMethod("PUT"), url);
    request.Headers.TryAddWithoutValidation("Accept", "application/json");
    request.Headers.TryAddWithoutValidation("Content-Type", "application/json");
    request.Headers.TryAddWithoutValidation("Authorization", ql.token);
    Option option = new(ql.value, "JD_COOKIE", "京东Cookie", ql.id);
    request.Content = new StringContent(JsonConvert.SerializeObject(option));
    request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");
    var response = await httpClient.SendAsync(request);
    using HttpContent content = response.Content;
    JObject? res = JsonConvert.DeserializeObject(await content.ReadAsStringAsync()) as JObject;
    return res;
}

File: updateEnvs.rs
use reqwest::{get, header::HeaderMap, Client};
use serde_json::{Map, Value};
use std::collections::HashMap;
use std::time::Duration;

#[tokio::main]
async fn main() {
    let source_url = "";
    let source_client_id = "";
    let source_client_secret = "";

    let dist_url = "";
    let dist_client_id = "";
    let dist_client_secret = "";

    let source_token = get_token(source_url, source_client_id, source_client_secret)
        .await
        .replace('"', "");

    let cookie_info = get_envs(source_url, source_token.as_str()).await;
    let cookie_array = cookie_info["data"].as_array().unwrap();
    let mut cookies: Vec<String> = vec![];
    for item in cookie_array {
        cookies.push(item["value"].to_string().replace('"', ""));
    }
    let cookie_str = cookies.join("&");

    let dist_token = get_token(dist_url, dist_client_id, dist_client_secret)
        .await
        .replace('"', "");
    let dist_cookies_info = get_envs(dist_url, dist_token.as_str()).await;
    let dist_cookies_array = dist_cookies_info["data"].as_array().unwrap();

    let mut id = String::new();
    for item in dist_cookies_array {
        if item["name"].as_str().unwrap() == "JD_COOKIE" {
            id = item["id"].to_string();
        }
    }

    let res = update_envs(
        id.as_str(),
        dist_url,
        dist_token.as_str(),
        cookie_str.as_str(),
    )
    .await;

    if res["code"].to_string().contains("200") {
        println!("更新JD_COOKIE环境变量成功!");
    } else {
        println!("更新JD_COOKIE环境变量失败!");
    }
}

async fn get_token(url: &str, client_id: &str, client_secret: &str) -> String {
    let res = get(format!(
        "{}/open/auth/token?client_id={}&client_secret={}",
        url, client_id, client_secret
    ))
    .await
    .unwrap()
    .json::<Map<String, Value>>()
    .await
    .unwrap();

    format!("{} {}", res["data"]["token_type"], res["data"]["token"])
}

async fn get_envs(url: &str, token: &str) -> Map<String, Value> {
    let mut headers = HeaderMap::new();
    headers.insert("Accept", "application/json".parse().unwrap());
    headers.insert("Authorization", token.parse().unwrap());

    let client = Client::new();
    client
        .get(format!("{}/open/envs", url))
        .headers(headers)
        .timeout(Duration::from_secs(3))
        .send()
        .await
        .unwrap()
        .json::<Map<String, Value>>()
        .await
        .unwrap()
}

async fn update_envs(id: &str, url: &str, token: &str, value: &str) -> Map<String, Value> {
    let mut headers = HeaderMap::new();
    headers.insert("Accept", "application/json".parse().unwrap());
    headers.insert("Content-Type", "application/json".parse().unwrap());
    headers.insert("Authorization", token.parse().unwrap());

    let mut data = HashMap::new();
    data.insert("value", value);
    data.insert("name", "JD_COOKIE");
    data.insert("remarks", "京东Cookie");
    data.insert("id", id);

    let client = Client::new();
    client
        .put(format!("{}/open/envs", url))
        .headers(headers)
        .json(&data)
        .send()
        .await
        .unwrap()
        .json::<Map<String, Value>>()
        .await
        .unwrap()
}
